File: 프로그래머스/dfs_bfs/네트워크.py
def solution(n, computers):
    visited = [0 for _ in range(n)]
    answer = 0
    
    def dfs(i):
      visited[i] = 1

      for c in range(n):
        if i != c and computers[i][c] == 1 and visited[c] == 0:
          dfs(c)          
          
    for i in range(n):
      if visited[i] == 0:
        dfs(i)
        answer += 1
        
    return answer
  
print(solution(3, [[1, 1, 0], [1, 1, 0], [0, 0, 1]]))
print(solution(3, [[1, 1, 0], [1, 1, 1], [0, 1, 1]]))

File: 프로그래머스/dfs_bfs/네트워크.js
function dfs(i, computers, visited) {
  if (visited[i]) return 0;

  visited[i] = 1;
  for (let j in computers[i]) {
    if (computers[i][j]) {
      dfs(j, computers, visited);
    }
  }

  return 1;
}

function solution(n, computers) {
  const visited = Array(n).fill(0);
  let answer = 0;

  for (let i in computers[0]) {
    answer += dfs(i, computers, visited);
  }

  return answer;
}
